### PR TITLE
Feat/convert trait

### LINKED ISSUE #522: Replace `XLike` traits with a single conversion trait
## Description

We can replace all the various `XLike` traits with a single one that can be used instead of `TryInto`, using the crate `Result` rather than an associated error type.

Something like this:

```rust
trait ConvertTo<T> {
  fn convert(self) -> crate::Result<T>;

  fn convert_unchecked(self) -> T;
}
```

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This replaces various `XLike` traits with a single unified `ConvertTo`.

## Links to any relevant issues
- Closes #522

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

**Review by qrayven — DISMISSED** [review]

File: sdk/src/types/mod.rs [hunk 1]
```diff
@@ -8,3 +8,4 @@ extern crate std;
 
 pub mod api;
 pub mod block;
+pub mod convert;
```
Comment (thibault-martinez) [inline sdk/src/types/mod.rs]: This should probably be in `sdk/utils` instead?

File: sdk/src/types/convert.rs
Comment (kwek20) [inline sdk/src/types/convert.rs:1]: Should we implement like Rc, Box and Option here? Or leave for when we need it

File: sdk/src/wallet/error.rs [hunk 2]
```diff
@@ -136,13 +135,6 @@ impl Serialize for Error {
     }
 }
 
-// Implements conversion from Infallible. This never happens, but is required by the compiler.
-impl From<Infallible> for Error {
```
Comment (kwek20) [inline sdk/src/wallet/error.rs:140]: 💯

**Review by kwek20 — COMMENTED** [review]
idk im jsut thinking that maybe the trait should impl Into and TryInto, and we use those methods?

File: sdk/src/types/convert.rs
Comment (DaughterOfMars) [reply sdk/src/types/convert.rs:1]: Probably just wait until we need it.

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Changelog 😬

**Review by thibault-martinez — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

**Pull request merged**